addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -310,6 +310,39 @@ def action_open_label_layout(self):
                 'default_move_quantity': 'move'},
         }
 
+    def action_merge(self):
+        if len(self.picking_type_id) > 1:
+            raise UserError(_('Batch/Wave transfers of different operation types cannot be merged.'))
+        if len(set(self.mapped('is_wave'))) > 1:
+            raise UserError(_('Batch transfers cannot be merged with wave transfers and vice versa.'))
+        if len(set(self.mapped('state'))) > 1:
+            raise UserError(_('Batch/Wave transfers of different states cannot be merged.'))
+        if self.state in ['done', 'cancel']:
+            raise UserError(_('You cannot merge done or cancelled batch/wave transfers.'))
```

Comment:
```suggestion
        if len(set(self.mapped('state'))) > 1:
            raise UserError(_('Batch/Wave transfers with different states cannot be merged.'))
        if self[0].state in ['done', 'cancel']:
            raise UserError(_('You cannot merge done or cancelled batch/wave transfers.'))
```

The `self.state` will currently throw an error since it expects a singleton. Might be useful to add a test that merging batches works as expected :wink:

You should probably also add a check that `if not self: return` to the beginning of this method now that we're doing index referencing